draft-ietf-ppm-dap.md

```diff
@@ -2632,52 +2573,171 @@ distinct aggregation parameters. If the batch has been queried with more than
+  (255)
+} BatchMode;
+~~~
+
+The time-interval batch mode is designed to support applications in which
+reports are collected into batches grouped by an interval of time. The Collector
+specifies a "batch interval" that determines the time range for reports included
+in the batch. For each report in the batch, the time at which that report was
+generated (see {{upload-flow}}) MUST fall within the batch interval specified by
+the Collector.
+
+Typically the Collector issues queries for which the batch intervals are
+continuous, monotonically increasing, and have the same duration. For example,
+the sequence of batch intervals `(1659544000, 1000)`, `(1659545000, 1000)`,
+`(1659546000, 1000)`, `(1659547000, 1000)` satisfies these conditions. (The
+first element of the pair denotes the start of the batch interval and the second
+denotes the duration.) However, this is not a requirement--the Collector may
+decide to issue queries out-of-order. In addition, the Collector may need to
+vary the duration to adjust to changing report upload rates.
+
+### Query Configuration
+
+The payload of `Query.config` is
+
+struct {
+  Interval batch_interval;
+} TimeIntervalQueryConfig;
+~~~
+
+where `batch_interval` is the batch interval requested by the Collector.
+
+### Partial Batch Selector Configuration
+
+The payload of `PartialBatchSelector.config` is empty.
+
+### Batch Selector Configuration
+
+The payload of `Query.config` is
+
+~~~ tls-presentation
+struct {
+  Interval batch_interval;
+} TimeIntervalBatchSelector;
+~~~
+
+where `batch_interval` is the batch interval requested by the Collector.
+
+### Batch Buckets {#time-interval-batch-buckets}
+
+Each batch bucket is identified by an interval of time whose beginning is a
+multiple of the task's `time_precision` and whose duration is equal to the
+task's `time_precision`. The identifier associated with a given report is the
+unique such interval containing the timestamp of the report; for example, if
+the task's `time_precision` is 1000 seconds and the report's timestamp is
+1729629081, the relevant batch bucket identifier is `(1729629000, 1000)`. The
+first element of the pair denotes the start of the batch interval and the
+second denotes the duration.
+
+The `Query` received by the Leader determines a valid set of batch bucket
+identifiers if the following conditions hold (likewise for the `BatchSelector`
+received by the Helper):
 
 * `batch_interval.duration >= time_precision` (this field determines,
   effectively, the minimum batch duration)
 
 * both `batch_interval.start` and `batch_interval.duration` are divisible by
   `time_precision`
 
-These measures ensure that Aggregators can efficiently "pre-aggregate" output
-shares recovered during the aggregation interaction.
+A batch consists of a sequence of contiguous batch buckets. That is, the set of
+batch buckets identifiers for the batch interval is
+`(batch_interval.start, batch_interval.start + batch_interval.duration)`,
```

Comment:
In this example, I think we want to replace `batch_interval.duration` with `time_precision` throughout. Also, I think the duration for all of the batch bucket identifiers should be `time_precision`.